homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -37,19 +37,22 @@ def __init__(
             handle_token=False,
             handle_rest=False,
         )
+        self.ws_connected: bool = False
 
     async def _async_update_data(self) -> None:
         """Subscribe for websocket and poll data from the API."""
-        await self.mowersession.connect()
-        self.mowersession.register_data_callback(
-            self.callback, schedule_immediately=True
-        )
+        if not self.ws_connected:
+            await self.mowersession.connect()
+            self.mowersession.register_data_callback(
```

Comment:
If this just stores the callback reference, we probably don't need to do it after every connect call, but just once for the coordinator. Or is the callback unregistered if the connection is lost?